src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -162,7 +162,18 @@ private IList<ApiParameterDescription> GetParameters(ApiParameterContext context
                 foreach (var actionParameter in context.ActionDescriptor.Parameters)
                 {
                     var visitor = new PseudoModelBindingVisitor(context, actionParameter);
-                    var metadata = _modelMetadataProvider.GetMetadataForType(actionParameter.ParameterType);
+
+                    ModelMetadata metadata = null;
+                    if (actionParameter is ControllerParameterDescriptor controllerParameterDescriptor &&
+                        _modelMetadataProvider is ModelMetadataProvider provider)
+                    {
+                        metadata = provider.GetMetadataForParameter(controllerParameterDescriptor.ParameterInfo);
+                    }
+
+                    if (metadata == null)
+                    {
+                        metadata = _modelMetadataProvider.GetMetadataForType(actionParameter.ParameterType);
```

Comment:
As [mentioned](https://github.com/aspnet/Mvc/issues/7435#issuecomment-370574332) in #7435, that bug may be a symptom of a broader issue. Need to check other places we call `GetMetadataForType(...)` before declaring #7435 resolved. Have you done that investigation?

For example, a couple of the `ParameterBinder.BindModelAsync(...)` overloads unconditionally call `GetMetadataForType(...)`. Since those overloads are not `[Obsolete]` and are in fact called from elsewhere in MVC, may get a subset of the relevant metadata in a few cases.

Another example: The usage in `HeaderModelBinderProvider` is likely fine but I may be ignoring a scenario.

Also look at where we already special-case `something is ControllerParameterDescriptor` to make sure we're not mishandling controller properties, page model properties, and / or page handler parameters.